### ISSUE TITLE
How to prefer a default classification for all functions?

### ISSUE THREAD

**Issue by resolritter** [kkinnear/zprint#155]
I've read that you can define a classification for a **single** specific function through `:fn-map`. But what if I want to have a default styling for all functions?

I've read the source code and am now guessing this is the part which would need to change

https://github.com/kkinnear/zprint/blob/5c52edb3ea1ac6e1ae14236dc334ad3e661c5c00/src/zprint/zprint.cljc#L3352

Or if it's possible already, somehow, please let me know. I've tried setting `:fn-style` at the root of the configuration map, but of course it didn't work

```
java.lang.Exception: set-options! for command-line options found these errors: In command-line options, In the key-sequence [:fn-style] the key :fn-style was not recognized as valid!.
```

**Comment by kkinnear** [comment]
Thanks for asking!  No, this isn't possible at present, simply because it never occurred to me to provide this capability.  You are correct, that is the area of the code that would need to change (along with the specs and maybe other stuff in the configuration validation).

In thinking about how I would implement something like what you describe, I can see two different places where it might be useful.

1. There could be a `:pre-default` "function" in the `:fn-map` which would specify a function-type and, possibly, an options map for every function __prior__ to their being looked up by name in the `:fn-map`.  If they weren't in the `:fn-map`, then this would be the function-type and options map for the function.  If they were in the `:fn-map`, then the function-type would change and any new options map would be merged __on top of__ the existing options map (which would contain the default options map value(s)).

2. There could be a `:post-default` function in the `:fn-map` which would have a function-type and an options map, and would only be used if the function to be formatted wasn't found in the `:fn-map`.  The function would be formatted with the function-type specified and the options map specified would be merged into the existing options map (also, possibly, __on top of__ any options map specified in the `:pre-default` `:fn-map` value).

I could see some value in either of these approaches, though using them together would seem a bit much.

Which, if either, of these capabilities would give you what you want?

Thanks for your feedback!

**Comment by resolritter** [comment]
The `:post-default` sounds less meddlesome because it implies precedence on `:fn-map`, from what I understood.  If the function is defined in `:fn-map`, it completely replaces the `:default` styling; and, conversely, the `:default` styling is only applied if the function is not specified in `:fn-map`

I'm not a veteran lisper so I can't evaluate sustainability on either approach, but I was thinking it could simply be an exclusion rule without any merging going on. Merging is sometimes troublesome because it means you need to "negate" everything that's not wanted, as opposed to define a new set of rules from scratch, which doesn't suffer from this problem.

Given this is a very niche request already, I think the functionality could start as straightforward and "dumb" as possible, which sounds less costly to implement. If the API is somehow limited, then it could be evolved as the use-cases come along.

**Comment by kkinnear** [comment]
I like your thinking on this, I'll see what I can do for something simple.  Thanks!

**Comment by kkinnear** [comment]
Implemented in 1.0.1.

**Comment by resolritter** [comment]
Works great! Thanks :D
